app/controllers/admin/config/about_controller.rb

```diff
@@ -5,5 +5,44 @@ def index
   end
 
   def update
+    settings_map = {}
+    if general_settings = params[:general_settings]
+      settings_map[:title] = general_settings[:name]
+      settings_map[:site_description] = general_settings[:summary]
+      settings_map[:about_banner_image] = general_settings[:about_banner_image]
+
+      settings_map[:extended_site_description] = general_settings[:extended_description]
+      if settings_map[:extended_site_description].present?
+        settings_map[:extended_site_description_cooked] = PrettyText.markdown(
+          settings_map[:extended_site_description],
+        )
+      else
+        settings_map[:extended_site_description_cooked] = ""
+      end
+    end
+
+    if contact_information = params[:contact_information]
+      settings_map[:community_owner] = contact_information[:community_owner]
+      settings_map[:contact_email] = contact_information[:contact_email]
+      settings_map[:contact_url] = contact_information[:contact_url]
+      settings_map[:site_contact_username] = contact_information[:contact_username]
+      settings_map[:site_contact_group_name] = contact_information[:contact_group_name]
+    end
+
+    if your_organization = params[:your_organization]
+      settings_map[:company_name] = your_organization[:company_name]
+      settings_map[:governing_law] = your_organization[:governing_law]
+      settings_map[:city_for_disputes] = your_organization[:city_for_disputes]
+    end
+
+    settings_map.each do |name, value|
+      with_service(
```

Comment:
We need to be handling the `on_success` and `on_failed_policy` checks for this service:

https://github.com/discourse/discourse/blob/48c58499f1399baf88b43762d39bb0849eb00cf0/app/controllers/admin/site_settings_controller.rb#L45-L61

We also should write a request spec to make sure that doing this in a loop is okay and that failure is handled correctly.

Reply:
I don't know what makes sense to do inside `on_success` here? Calling `SiteSettingUpdateExistingUsers` here would essentially be a no-op because it only applies to specific site settings which we don't touch here.

It's also not possible for the `setting_is_visible` or `setting_is_configurable` policies to fail here because we change specific settings here none of which are hidden or unconfigurable (the ones that are we will `allow_changing_hidden: true` for them).

Reply:
I think just move `render json: success_json` into `on_success`, then add a general `on_failure` call like this chat endpoint to handle anything else:

https://github.com/discourse/discourse/blob/e92a82aa1d16e24a9447930d0d3441716b40efaa/plugins/chat/app/controllers/chat/api/channels_controller.rb#L39-L40

Main point is we need to handle the `on_X` stuff explicitly

Reply:
Hmm I just realized that because I'm currently calling the service for each setting inside a loop, I can't do any `render`ing within service callbacks because it's not allowed (nor does it make sense) to `render` multiple times in an action. Tricky stuff...

Reply:
Ah right that is a pickle :thinking: I think really the ultimate fix would be to make a service that updates multiple site settings to new values. This way you could run the policies once for all settings, and the success/failure would only be once too since you are updating all the settings (and adding logs) in a single transaction.

For the sake of getting this PR over the line, since it's still hidden behind various settings, let's punt on this for now. I would like you to start thinking of this service though. You will likely need this other feature of services, where you can give a separate class to a policy:

https://github.com/discourse/discourse/blob/0b3b101c97b6a027ee11bcd166d1ab49a63a0c00/plugins/chat/app/services/chat/create_direct_message_channel.rb#L29-L30

https://github.com/discourse/discourse/blob/0b3b101c97b6a027ee11bcd166d1ab49a63a0c00/plugins/chat/app/policies/chat/direct_message_channel/max_users_excess_policy.rb